avocado/core/test.py

```diff
@@ -337,13 +429,74 @@ def datadir(self):
         host data files that are generic enough to be used for all tests
         contained in a given test file.
         """
+        # TODO: on removal of Test.datadir, move its code over to
+        # get_file_data_dir
         # Maximal allowed file name length is 255
         if (self.filename is not None and
                 len(os.path.basename(self.filename)) < 251):
```

Comment:
looking at this I think we should use `astring.string_to_safe_path` to avoid constants like `251` being present on various places... Do you want take care of that (to avoid clashes)?

Reply:
Yep, I agree.

Reply:
Btw feel free to address this or just create a card so we can fix it when this job is done...